### ISSUE TITLE
String contents is being json formatted

### ISSUE THREAD

**Issue by powellquiring** [lydell/json-stringify-pretty-compact#4]
Sorry to give such a poorly written use case.  Maybe I can get this into a testcase in the future but I wanted to capture the issue before I forgot about it.

I ran into a case where the value of an object was a string that contained some valid json and it was formatted.

Notice the difference in the spacing in this portion of the comments string: {errors:insufficient_storage,status:507}

Here is a portion of the original input:

```
 {
    "_id": "58d421ee0cfbd700344e4c6c",
    "comments": "Could not instantiate with error -Service broker error: {+description\"=>Deployment creation failed - {errors:insufficient_storage,status:507}+ }-",
    "rating": 1,
    "criteria": "Payment",
    "offering": "Compose for RabbitMQ",
    "service": "Message Queue",
    "platform": "Bluemix",
    "author": "xxx@foo.com",
    "type": "rating",
    "__v": 0,
    "date": "2017-03-16T00:00:00.000Z"
  },

```

Here is a portion of the resulting stringify output:

```
{
 "author": "",
 "date": "",
 "offerings": [
  {
   "old": [
    {"_id": "58d421ee0cfbd700344e4c6c", "comments": "Could not instantiate with error -Service broker error: {+description\"=>Deployment creation failed - {errors: insufficient_storage, status: 507}+ }-","rating":1,"criteria":"Payment","offering":"Compose for RabbitMQ","service":"Message Queue","platform":"Bluemix","author":"xxx@foo.com","type":"rating","__v":0,"date":"2017-03-16T00: 00: 00.000Z"},
    {"_id": "58d421ee0cfbd700344e4c62", "comments": "Could not instantiate with error -Service broker error: {+description\"=>Deployment creation failed - {errors: insufficient_storage, status: 507}+ }-","rating":1,"criteria":"Security","offering":"Compose for RabbitMQ","service":"Message Queue","platform":"Bluemix","author":"xxx@foo.com","type":"rating","__v":0,"date":"2017-03-16T00: 00: 00.000Z"},
```

**Comment by lydell** [comment]
Hi! Thanks for the issue! Unfortunately, I can't reproduce this.

test.json:

```json
{
   "_id": "58d421ee0cfbd700344e4c6c",
   "comments": "Could not instantiate with error -Service broker error: {+description\"=>Deployment creation failed - {errors:insufficient_storage,status:507}+ }-",
   "rating": 1,
   "criteria": "Payment",
   "offering": "Compose for RabbitMQ",
   "service": "Message Queue",
   "platform": "Bluemix",
   "author": "xxx@foo.com",
   "type": "rating",
   "__v": 0,
   "date": "2017-03-16T00:00:00.000Z"
 }
```

```
$ node
> fs.writeFileSync("output.json", require("json-stringify-pretty-compact")(require("./test.json")))
```

output.json:

```json
{
  "_id": "58d421ee0cfbd700344e4c6c",
  "comments": "Could not instantiate with error -Service broker error: {+description\"=>Deployment creation failed - {errors:insufficient_storage,status:507}+ }-",
  "rating": 1,
  "criteria": "Payment",
  "offering": "Compose for RabbitMQ",
  "service": "Message Queue",
  "platform": "Bluemix",
  "author": "xxx@foo.com",
  "type": "rating",
  "__v": 0,
  "date": "2017-03-16T00:00:00.000Z"
}
```

**Comment by powellquiring** [comment]
Try this test:

testStringify('{"{s:0}}', '"{\\"{s:0}}"')

**Comment by lydell** [comment]
Awesome, thanks! Fixed in 1.0.4.
